macros/core/PGbasicmacros.pl

```diff
@@ -2995,18 +3006,24 @@ sub image {
 				display_options2(%known_options);
 		}
 	}
-	my $width       = $out_options{width};
-	my $height      = $out_options{height};
-	my $tex_size    = $out_options{tex_size};
+
+	# Get options for width, height, and tex_size, with a sanity check for integer values.
+	my $width    = $out_options{width}    =~ /[1-9]\d*/ ? $out_options{width}    : '';
+	my $height   = $out_options{height}   =~ /[1-9]\d*/ ? $out_options{height}   : '';
+	my $tex_size = $out_options{tex_size} =~ /[1-9]\d*/ ? $out_options{tex_size} : '';
```

Comment:
I realized that these sanity checks aren't quite right.  They should be `/^[1-9]\d*$/`.  If the regex is not anchored, then something like 'asdf500' would match and cause a warning.

Reply:
Good catch, updated now.